src/transformers/modeling_utils.py

```diff
@@ -1809,7 +1818,15 @@ def save_pretrained(
         weights_name = SAFE_WEIGHTS_NAME if safe_serialization else WEIGHTS_NAME
         weights_name = _add_variant(weights_name, variant)
 
-        shards, index = shard_checkpoint(state_dict, max_shard_size=max_shard_size, weights_name=weights_name)
+        # For now only 8bit models contains metadata on the model's state_dict
+        state_dict_contains_metadata = getattr(self, "is_loaded_in_8bit", False)
```

Comment:
can we infer if we are dealing with 8bit from the state dict itself (I mean from the meta data in the new format) ?

Reply:
Yes, this seems hacky - having to check here and then add an extra in the `shard_checkpoint` method. Two questions: 
* What is the extra metadata stored in `state_dict`? 
* Would any weight be stored as a string otherwise in `state_dict`? i.e. would checking for `isinstance(weight, str)` be enough? If it is a string, then it would fail the rest of the for-loop logic but I guess we might want to catch this failure if it happens outside of bnb?

Reply:
> What is the extra metadata stored in state_dict?

Some weights are stored as pure strings

> i.e. would checking for isinstance(weight, str) be enough? 

I think it won't happen outside the scope of bnb modules as in transformers we only store torch tensors inside state dicts, hence the check `state_dict_contains_metadata = getattr(self, "is_loaded_in_8bit", False)`. 

What do you think?

Reply:
For me, it would be better not to introduce `state_dict_contains_metadata`, especially if it's not absolutely necessary (at this moment)

Reply:
Sounds good, reverted that